mcp/src/minecraft/com/elementfx/tvp/ad/block/BlockCustomWorkbench.java

```diff
@@ -0,0 +1,256 @@
+     */
+    public int damageDropped(IBlockState state)
+    {
+        return ((BlockCustomWorkbench.EnumType)state.getValue(TYPE)).getMetadata();
+    }
+
+    /**
+     * returns a list of blocks with the same ID, but different meta (eg: wood returns 4 blocks)
+     */
+    public void getSubBlocks(Item itemIn, CreativeTabs tab, List<ItemStack> list)
+    {
+        for (BlockCustomWorkbench.EnumType blockcustomworkbench$enumtype : BlockCustomWorkbench.EnumType.values())
+        {
+            list.add(new ItemStack(itemIn, 1, blockcustomworkbench$enumtype.getMetadata()));
+        }
+    }
+
+    /**
+     * Get the MapColor for this Block and the given BlockState
+     */
+    public MapColor getMapColor(IBlockState state)
+    {
+        return MapColor.WOOD;
+    }
+
+    /**
+     * Convert the given metadata into a BlockState for this Block
+     */
+    public IBlockState getStateFromMeta(int meta)
+    {
+        return this.getDefaultState().withProperty(TYPE, BlockCustomWorkbench.EnumType.byMetadata(meta));
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    public int getMetaFromState(IBlockState state)
+    {
+        return ((BlockCustomWorkbench.EnumType)state.getValue(TYPE)).getMetadata();
+    }
+    
+    public static String getName(IBlockState state)
+    {
+    	return ((BlockCustomWorkbench.EnumType)state.getValue(TYPE)).getName();
+    }
+
+    protected BlockStateContainer createBlockState()
+    {
+        return new BlockStateContainer(this, new IProperty[] {TYPE});
+    }
+    
+    public boolean isOpaqueCube(IBlockState state) 
+    {
+    	return false;
+    }
+    
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.CUTOUT;
+    }
+    
+    public boolean isFullCube(IBlockState state)
+    {
+        return false;
+    }
+    
+    public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, @Nullable ItemStack heldItem, EnumFacing side, float hitX, float hitY, float hitZ)
+    {
+        if (worldIn.isRemote)
+        {
+            return true;
+        }
+        else
+        {
+            playerIn.displayGui(new BlockCustomWorkbench.InterfaceCustomCraftingTable(worldIn, pos));
+            playerIn.addStat(StatList.CRAFTING_TABLE_INTERACTION);
+            return true;
+        }
+    }
+    
+    public static class InterfaceCustomCraftingTable extends InterfaceCraftingTable
+    {
+        private final World world;
+        private final BlockPos position;
+
+        public InterfaceCustomCraftingTable(World worldIn, BlockPos pos)
+        {
+        	super(worldIn, pos);
+            this.world = worldIn;
+            this.position = pos;
+        }
+
+        public ITextComponent getDisplayName()
+        {
+            return new TextComponentTranslation("tile.customWorkbench.elven.name", new Object[0]);
```

Comment:
I'm not sure where if anywhere this is used, but it doesn't seem right to leave it hard-coded to the elven workbench. Can we do something like `"tile.customWorkbench." + BlockCustomWorkbench.getUnlocalizedName(world.getBlockState(position)) + ".name"` with the getUnlocalizedName function as:
```
public static String getUnlocalizedName(IBlockState state)
{
    return ((BlockCustomWorkbench.EnumType)state.getValue(TYPE)).getUnlocalizedName();
}
```